src/client/app/common/views/components/signin.vue

```diff
@@ -27,6 +45,12 @@ import i18n from '../../../i18n';
 import { apiUrl, host } from '../../../config';
 import { toUnicode } from 'punycode';
 
+function hexifyAB(buffer) {
+	return Array.from(new Uint8Array(buffer))
+		.map(item => item.toString(16).padStart(2, 0))
+		.join('');
+}
```

Comment:
If you use this function from multiple files, consider moving the code to a dedicated file rather than copying and pasting.

Reply:
@rinsuki Where should I put the file? I'm not really sure where that sort of thing fits within the structure of the app :/